src/emulator/storage/persistence.ts

```diff
@@ -2,12 +2,19 @@ import { openSync, closeSync, readSync, unlinkSync, renameSync, mkdirSync } from
 import * as rimraf from "rimraf";
 import * as fs from "fs";
 import * as path from "path";
+import * as uuid from "uuid";
 
-/** Helper for disk I/O operations. */
+/**
+ * Helper for disk I/O operations.
+ * Assigns a unique identifier to each file and stores it on disk based on that identifier
+ */
 export class Persistence {
   private _dirPath!: string;
+  // Mapping from emulator filePaths to unique identifiers on disk
+  private _diskPathMap: Map<string, string>;
```

Comment:
I'm a little concerned that we're not consistent in how we're using maintaining `diskPathMap`, i.e. deleting a file on disk doesn't remove the key from the map, any method using `getDiskPath` can either use the map or default to reading directly from the disk, etc.

If we're introducing virtual file pointers, can we stay consistent and avoid ever reading the absolute file path passed into these methods?

Reply:
You bring up some good points, I'll be a bit more careful in staying consistent and make sure the virtual file system is the only one in use